### PR TITLE
Change deploy process scripts to make it Linux friendly

### PR DESCRIPTION
**Pull request by andyinaus**
# Background

We are using Linux container to deploy a new release now.

## Issue(s) found while switching to a `Linux OS` are:
1. `$env:TEMP` is only available in `Windows OS`
**Solution:** Use cross-platform temp path `([System.IO.Path]::GetTempPath())` instead
2. `Linux OS` uses case sensitive file system whereas `Windows` does not.
**Solution:** Renaming files instead. 

**Note:** If you ever encountered any issue(s) with `FileNotFound` due to this renaming, please ensure all the files renamed in this PR are named correctly in your local/remote branch.

# How to review this PR

Quality ✔️ 

# Pre-requisites
- [x] I have considered informing or consulting the right people, e.g.: 
  - For changes to the execution pipeline: [`#topic-execution`](https://octopusdeploy.slack.com/archives/C3KT1DFSM)
  - For other changes, consider the other `#topic-...` channels
- [x] I have considered whether this should be a patch or wait for a minor.
- [x] I have considered appropriate testing for my change.

### CODE REVIEW

File: embed-octo.ps1 [hunk 1]
```diff
@@ -71,7 +71,7 @@ if(!(Test-Path $destinationFolder)){
     New-Item -ItemType Directory -Path $destinationFolder | Out-Null
 }
 
-$downloadFolder = Join-Path $env:TEMP "octo"
+$downloadFolder = Join-Path ([System.IO.Path]::GetTempPath()) "octo"
```
Comment (tothegills) [inline embed-octo.ps1:74]: 👍 cross platform temp path

File: pack.ps1 [hunk 2]
```diff
@@ -15,6 +15,9 @@ $ErrorActionPreference = "Stop"
 $buildDirectoryPath = "$basePath/dist"
 $buildArtifactsPath = "$buildDirectoryPath/Artifacts"
 
+<# This is to make extension-icon.$environment.png and extension-manifest.$environment.json Linux friendly since it has a case sensitive file system#>
```
Comment (tothegills) [inline pack.ps1:18]: Where does $environment come from?

Comment (andyinaus) [reply pack.ps1:18]: #{Octopus.Environment.Name}

Comment (andyinaus) [reply pack.ps1:18]: If we are okay to change them to lowercase then it may be a better option if there is no other side-effects

Comment (tothegills) [reply pack.ps1:18]: Was thinking along the lines of renaming the files so they match the case of the acceptable environments?

Comment (andyinaus) [reply pack.ps1]: That's another option

**Review by tothegills — APPROVED** [review]
Decided the change the files names to the case we are expecting.

**Pull request merged**